src/app/signup/actions.ts

```diff
@@ -0,0 +1,43 @@
+"use server";
+
+import { PrismaClient } from "@prisma/client";
+import bcrypt from "bcrypt";
+import { revalidatePath } from "next/cache";
+import { z } from "zod";
+
+const prisma = new PrismaClient();
+
+const signUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(8),
+  name: z.string().optional(),
+  surname: z.string().optional(),
+  companyName: z.string().optional(),
+});
+
+export async function signUpUser(formData: FormData) {
+  const formDataObj: Record<string, string | undefined> = {};
```

Comment:
nitpick: we usually avoid abbreviations. So it's preferable to just have `formDataObject`